ui: fix diagnostics expires after input validation

jasonlmfong opened the pull request:
currently the field for expires_after does not allow full deletion of the inputted number, leading to a poor experience

this is changed by allowing full deletions, while maintaining sanitation for integrity of integer outputs

demo: https://github.com/user-attachments/assets/3a98554b-6d02-44ce-8019-796f3c7a7339

Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/154251)
<!-- Reviewable:end -->

kyle-a-wong commented on pkg/ui/workspaces/cluster-ui/src/statementsDiagnostics/activateStatementDiagnosticsModal.tsx:
```diff
@@ -293,8 +293,10 @@ export const ActivateStatementDiagnosticsModal = React.forwardRef<
               value={expiresAfter}
               onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
-                if (parseInt(e.target.value) > 0) {
+                if (!isNaN(parseInt(e.target.value))) {
```
It seems like this change will allow for negative numbers. Is this something we want to support?

jasonlmfong committed:
ui: fix diagnostics expires after input validation

currently the field for expires_after does not allow full deletion of the inputted number, leading to a poor experience

this is changed by allowing full deletions, while maintaining sanitation for integrity of integer outputs

Epic: None
Release note: None

kyle-a-wong reviewed (approved):
lgtm

jasonlmfong commented:
bors r+

jasonlmfong commented:
bors retry
